src/Dependencies/Collections/ImmutableSegmentedDictionary`2.cs

```diff
@@ -0,0 +1,360 @@
+        }
+
+        public bool ContainsKey(TKey key)
+            => _dictionary.ContainsKey(key);
+
+        public bool ContainsValue(TValue value)
+            => _dictionary.ContainsValue(value);
+
+        public Enumerator GetEnumerator()
+            => new(_dictionary, Enumerator.ReturnType.KeyValuePair);
+
+        public ImmutableSegmentedDictionary<TKey, TValue> Remove(TKey key)
+        {
+            var self = this;
+            if (!self._dictionary.ContainsKey(key))
+                return self;
+
+            var dictionary = new SegmentedDictionary<TKey, TValue>(self._dictionary, self._dictionary.Comparer);
+            dictionary.Remove(key);
+            return new ImmutableSegmentedDictionary<TKey, TValue>(dictionary);
+        }
+
+        public ImmutableSegmentedDictionary<TKey, TValue> RemoveRange(IEnumerable<TKey> keys)
+        {
+            if (keys is null)
+                throw new ArgumentNullException(nameof(keys));
+
+            var result = ToBuilder();
+            result.RemoveRange(keys);
+            return result.ToImmutable();
+        }
+
+        public ImmutableSegmentedDictionary<TKey, TValue> SetItem(TKey key, TValue value)
+        {
+            var self = this;
+            var dictionary = new SegmentedDictionary<TKey, TValue>(self._dictionary, self._dictionary.Comparer);
+            dictionary[key] = value;
+            return new ImmutableSegmentedDictionary<TKey, TValue>(dictionary);
+        }
+
+        public ImmutableSegmentedDictionary<TKey, TValue> SetItems(IEnumerable<KeyValuePair<TKey, TValue>> items)
+        {
+            if (items is null)
+                throw new ArgumentNullException(nameof(items));
+
+            var result = ToBuilder();
+            foreach (var item in items)
+            {
+                result[item.Key] = item.Value;
+            }
+
+            return result.ToImmutable();
+        }
+
+#pragma warning disable CS8767 // Nullability of reference types in type of parameter doesn't match implicitly implemented member (possibly because of nullability attributes).
+        public bool TryGetKey(TKey equalKey, [NotNullWhen(true)] out TKey? actualKey)
+#pragma warning restore CS8767 // Nullability of reference types in type of parameter doesn't match implicitly implemented member (possibly because of nullability attributes).
+        {
+            var self = this;
+            foreach (var key in self.Keys)
+            {
+                if (self.KeyComparer.Equals(key, equalKey))
+                {
+                    actualKey = key;
+                    return true;
+                }
+            }
+
+            actualKey = default;
+            return false;
+        }
+
+#pragma warning disable CS8767 // Nullability of reference types in type of parameter doesn't match implicitly implemented member (possibly because of nullability attributes).
+        public bool TryGetValue(TKey key, [MaybeNullWhen(false)] out TValue value)
```

Comment:
>MaybeNullWhen [](start = 43, length = 13)

I'd rather make this attribute conditional on `NETCOREAPP`.

Reply:
➡️ This is the correct signature for this method, regardless of what the interface says.

Reply:
But `MaybeNullWhen` is not available in projects that target net472. We shouldn't force them to define these attribute.

Reply:
📝 It appears that this suppression is no longer necessary

Reply:
➡️ Suppression now removed